v2/protocol/http/retries_result.go

```diff
@@ -0,0 +1,46 @@
+package http
+
+import (
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/protocol"
+)
+
+// NewRetriesResult returns a http RetriesResult that should be used as
+// a transport.Result without retries
+func NewRetriesResult(result protocol.Result, retries int, results []protocol.Result) protocol.Result {
+	return &RetriesResult{
+		Result:  result,
+		Retries: retries,
+		Results: results,
+	}
+}
+
+// Result wraps the fields required to make adjustments for http Responses.
+type RetriesResult struct {
+	// The last result
+	protocol.Result
+
+	// Retries is the number of times the request was tried
+	Retries int
```

Comment:
I think we want to return back the total time spent trying too if we do exp backoff, what do you think?

Reply:
to make sure we are on the same page, this is the time spent trying *excluding* the succesful (if any) request, right. I'm pretty sure the answer is yes...